nemo/deploy/nlp/megatronllm_deployable.py

```diff
@@ -99,6 +103,152 @@ def to_long_tensor(self):
         return torch.tensor([self], dtype=torch.long, device='cuda')
 
 
+class MegatronLLMDeploy:
+
+    @staticmethod
+    def get_deployable(
+        nemo_checkpoint_filepath: str = None,
+        num_devices: int = 1,
+        num_nodes: int = 1,
+        tensor_model_parallel_size: int = 1,
```

Comment:
Params `num_devices` & `num_nodes` are quite tied to `tensor_model_parallel_size` & `pipeline_model_parallel_size`.

How about making ` tensor_model_parallel_size` & `pipeline_model_parallel_size` optional and setting them by default to `num_devices` & `num_nodes`, respectitvely (if they are `None`)?

Some assertion for consistency like ` num_devices * num_nodes == tensor_model_parallel_size * pipeline_model_parallel_size`  would also be useful.

(Not sure how `context_parallel_size` changes all this).